java-frontend/src/main/java/org/sonar/java/resolve/BytecodeVisitor.java

```diff
@@ -305,11 +306,14 @@ public void visitEnd() {
     if (bytecodeNames == null) {
       return ImmutableList.of();
     }
-    ImmutableList.Builder<JavaType> types = ImmutableList.builder();
+    List<JavaType> types = new ArrayList<>();
     for (String bytecodeName : bytecodeNames) {
-      types.add(getClassSymbol(bytecodeName).type);
+      JavaSymbol.TypeJavaSymbol completedClassSymbol = getClassSymbol(bytecodeName);
+      if(!completedClassSymbol.isUnknown()) {
```

Comment:
We should not remove unknown exception types, or we will fail on replacing thrown types later with `IndexOutOfBoundExceptions`.